design/src/rfcs/rfc00014_using_features_responsibly.md

```diff
@@ -0,0 +1,172 @@
+    }
+}
+```
+
+When the example code above is compiled with both features enabled, compilation will fail with a "duplicate definitions with name `tls_adapter`" error. Also, note that the return type of the function differs between the two versions. This is a SemVer-incompatible change.
+
+Here's an updated version of the example that fixes these issues:
+
+```rust
+#[cfg(feature = "rustls")]
+impl<M, R> ClientBuilder<(), M, R> {
+    /// Connect to the service over HTTPS using Rustls.
+    pub fn rustls(self) -> ClientBuilder<Adapter<crate::conns::Https>, M, R> {
+        self.connector(Adapter::builder().build(crate::conns::https()))
+    }
+}
+
+#[cfg(feature = "native-tls")]
+impl<M, R> ClientBuilder<(), M, R> {
+    /// Connect to the service over HTTPS using the native TLS library on your platform.
+    pub fn native_tls(
+        self,
+    ) -> ClientBuilder<Adapter<hyper_tls::HttpsConnector<hyper::client::HttpConnector>>, M, R> {
+        self.connector(Adapter::builder().build(crate::conns::native_tls()))
+    }
+}
+```
+
+Both features can now be enabled at once without creating a conflict. Since both methods have different names, it's now Ok for them to have different return types.
+
+[*This is real code, see it in context*](https://github.com/awslabs/smithy-rs/blob/2e7ed943513203f1472f2490866dc4fb8a392bd3/rust-runtime/aws-smithy-client/src/hyper_ext.rs#L303)
+
+### We should avoid using `#[cfg(not(feature = "some-feature"))]`
+
+At the risk of seeming repetitive, the Cargo book says:
+
+> enabling a feature should not disable functionality, and it should usually be safe to enable any combination of features
+
+Conditionally compiling code when a feature is **not** activated can make it hard for users and maintainers to reason about what will happen when they activate a feature. This is also a sign that a feature may not be "additive".
+
+***NOTE***: It's ok to use `#[cfg(not())]` to conditionally compile code based on a user's OS. It's also useful when controlling what code gets rendered when testing or when generating docs.
+
+One case where using `not` is acceptable is when providing a fallback when no features are set:
+
+```rust
+#[cfg(feature = "rt-tokio")]
+pub fn default_async_sleep() -> Option<Arc<dyn AsyncSleep>> {
+    Some(sleep_tokio())
+}
+
+#[cfg(not(feature = "rt-tokio"))]
+pub fn default_async_sleep() -> Option<Arc<dyn AsyncSleep>> {
+    None
+}
+```
+
+In this example, code that calls the `default_async_sleep` function will continue to compile when the `rt-tokio` feature is inactive. This moves the responsibility of handling "missing" functionality out of the places where that functionality is called upon into the place where that functionality is defined. This results in less redundant code. However, once more than one mutually exclusive feature is added, it's better to use the [`cfg!` macro][cfg! macro] with an if-else statement:
+
+```rust
+pub fn default_async_sleep() -> Option<Arc<dyn AsyncSleep>> {
```

Comment:
The "When a default feature providing some functionality is disabled, active features must not automatically replace that functionality" section below basically says don't do this, so maybe we just want to remove this example?